cfme/utils/template/base.py

```diff
@@ -254,6 +262,34 @@ def download_image(self):
         else:
             return True
 
+    @log_wrap('unzip image')
+    def unzip_image(self, archive_type='zip'):
+        """
+        Unzips image and then changes image name to extracted one.
+
+        For EC2 and SCVMM images is zip used and for GCE is tar.gz used.
+        """
+        zip_path = self.image_name
+        try:
+            if archive_type == 'zip':
```

Comment:
I would think you can detect the archive type from the raw_image_url, just by looking at the file extension?  Then this doesn't have to take any arguments.

Reply:
Yes, it makes sense. I had problems with tar.gz so I didn't do it this way, but reworked.